src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -405,6 +412,18 @@ void AutomatedRssDownloader::on_removeRuleBtn_clicked()
         RSS::AutoDownloader::instance()->removeRule(item->text());
 }
 
+void AutomatedRssDownloader::on_addCategoryBtn_clicked()
+{
+    QString prevCategoryName = m_ui->comboCategory->currentText();
+    QString newCategoryName = TorrentCategoryDialog::createCategory(this);
+    AutomatedRssDownloader::loadCategoryList();
```

Comment:
Why reload entire list? Why not just add new category to it?

Reply:
@glassez The new category name should be inserted into the right place in sorted order. I just reloaded because it was more convenient as `loadCategoryList` already sorts the list. In terms of time complexity this would be slower. I will change to inserting to the preloaded list if that seems more reasonable.

Reply:
@glassez I've been looking into sorting `QComboBoxes` and learned that using `QSortFilterProxyModel` is a proper way. There is already a wrapper for it in `gui/categoryfilterproxymodel`, so I'll look into how to use it tomorrow from reading `gui/categoryfitlerwidget`.

Reply:
I don't think this should be a problem if you just add new Category to the beginning or end of the list, because it is assumed that the user will add a new category to use it immediately, so user is unlikely to be interested in its (temporary) position in the list.

Reply:
Hmm you're right I was trying to over-engineer this.